src/templates/_components/fieldtypes/Number/input.twig

```diff
@@ -8,7 +8,7 @@
 {%- set hasSuffix = suffix is not same as(false) %}
 
 {% if formatNumber %}
-    {{ hiddenInput("#{field.handle}[locale]", craft.app.formattingLocale.id) }}
+    {{ hiddenInput("#{field.handle}[locale]", currentUser.getPreferredLocale()) }}
```

Comment:
If these are different (and the user does in fact have a formatting locale set), that’s the bug.